### PR TITLE
Fix confidence function

### PR DESCRIPTION
**Pull request by szarsti**
Somehow this makes it work ok, while previous implementation caused data missalignment

### CODE REVIEW

**Comment by szarsti** [comment]
Before this change, for `SELECT m0, derivate(m0), confidence(m0) FROM 'c3bcee12-0680-4bf3-8237-f51b48330dd8'.'base'.'cpu' BUCKET 'c3' AS m0 LAST 2m` I was seeing results:
screen shot 2016-05-11 at 18 00 45

After it, I got right chart:
screen shot 2016-05-11 at 18 00 55

I am not quite sure why my change fixes the problem. I am a bit afraid that there is problem maybe somewhere else with realize/derealize function on some unusual data. Maybe my change just avoids real problem elsewhere by producing a bit cleaner data.

I would like to have my change included, because it avoid memory allocation on stack for variable that is used as static.

**Comment by Licenser** [comment]
Hi mate,
I'll try to wrap my head around why this works better before merging it I'd really like to know what the root cause is before merging something that magically fixes it :)

File: c_src/trans_nif.c [hunk 1]
```diff
@@ -89,7 +89,7 @@ derivate(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
   if (count < 1) // can't be empty
     return enif_make_badarg(env);
 
-  if (! (target = (decimal*) enif_make_new_binary(env, bin.size - sizeof(decimal), &r)))
+  if (! (target = (decimal*) enif_make_new_binary(env, (count - 1) * sizeof(decimal), &r)))
```
Comment (Licenser) [inline c_src/trans_nif.c]: I think this makes sense, the derivate will always be 1 element shorter, `bin.size` was probably a bug here.

File: c_src/trans_nif.c [hunk 2]
```diff
@@ -116,8 +115,18 @@ confidence(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
...
+    if (vs[i].confidence != 0)
+      target[i] = (decimal){
+        .coefficient = vs[i].confidence,
+        .exponent = -3,
+        .confidence = CERTAIN
+      };
+    else
+      target[i] = (decimal){
+        .coefficient = 0,
+        .exponent = 0,
```
Comment (Licenser) [inline c_src/trans_nif.c]: is .coefficent = 0 , .exponen = -3 going to cause a problem or is there another reason to expand this in two defintions?

Comment (szarsti) [inline c_src/trans_nif.c]: Yes, .exponent =-3 was causing problem when it had coefficient = 0, but I am still not sure why.

**Comment by szarsti** [comment]
Probably you are right, to spend some time investigating. It probably is just a cover of some more serious issue. I have run some additional test, and `rebar3 as eqc eqc -n 1000` crashes with segfault. It happens only if you run quickcheck with huge sample count. It could be that there is still some bug left that is causing adverse effect and memory access in some exotic data set.

I am investigating, because it may be crashing out frontend processes.

**Comment by szarsti** [comment]
Thank you for big help with getting that error tracked. That should have fixed it properly.

**Comment by Licenser** [comment]
awesome!

**Pull request merged**